### PR TITLE
Use of Constant TYPO3_cliMode

### PR DESCRIPTION
**Pull request by woodyc79**
Use Constant TYPO3_cliMode only in TYPO3 versions lower 8

### CODE REVIEW

**Comment by kaystrobach** [comment]
@woodyc79 and @CDRO can you please also verify that

**Comment by CDRO** [comment]
I get the `$isCliRequest = false` leads to an error, since it is indeed not used anywhere, but the static request to `GeneralUtilty` cannot be avoided I guess :-)

File: Classes/Service/DyncssService.php [hunk 1]
```diff
@@ -43,13 +43,22 @@ public static function getCompiledFile($inputFile)
     protected static function fixPathForInput($file)
     {
         if (TYPO3_MODE === 'FE') {
```
Comment (woodyc79) [inline Classes/Service/DyncssService.php]: Maybe throwing error if $file is empty/null or at least return something to identify that $file must not be empty (or set a default value $file="" in the parameters)

**Comment by kaystrobach** [comment]
needs testing and some refactoring later ... static calls should be avoided

File: Classes/Service/DyncssService.php (hunk 1 above)
Comment (CDRO) [reply Classes/Service/DyncssService.php]: I think the `\empty` throws an error, the PR checker should be updated accordingly I think.

**Comment by kaystrobach** [comment]
this extension definitely needs some unit tests 😄

**Comment by CDRO** [comment]
This solved the issues on TYPO3 8, I've sadly got no 7.6 with dyncss laying around to test it though.

**Comment by kaystrobach** [comment]
tests on 7.6 are green.

**Pull request merged**